src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1573,6 +1576,430 @@ LedgerManagerImpl::prefetchTransactionData(AbstractLedgerTxnParent& ltx,
     }
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+
+ThreadEntryMap
+LedgerManagerImpl::collectEntries(AbstractLedgerTxn& ltx, Thread const& txs)
+{
+    ThreadEntryMap entryMap;
+    auto getEntries = [&](TransactionFrameBasePtr tx,
```

Comment:
I've left some comments regarding this implementation below, but I think this whole method might benefit from factoring out the operation-specific logic into the operations, i.e. add a function like 'preloadEntries' or something like that, and add the respective metrics/error handling there. If there is too much duplication, maybe consider factoring out the common logic into a separate function as well.